### DISCUSSION TITLE
Running `npx eslint {path}` results in command failure.

### DISCUSSION THREAD

**Discussion by jeffhou** [eslint/eslint#14083] [Q&A]
**Tell us about your environment**

* Node version: v14.15.3
* npm version: v7.5.2
* Local ESLint version: v7.19.0 (Currently used)
* Global ESLint version: Not found
* OS - macOS 11.1 (M1)

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  'env': {
    'browser': true,
    'commonjs': true,
    'es2021': true,
    'node': true,
  },
  'extends': [
    'plugin:vue/essential',
    'google',
  ],
  'parserOptions': {
    'ecmaVersion': 12,
  },
  'plugins': [
    'vue',
  ],
  'rules': {
  },
};

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

I installed ESLint using `npm install eslint`, set up ESLint using `npx eslint --init`, and then ran `npx eslint README.md`.

```bash
npx eslint README.md
```

**What did you expect to happen?**
I expected it the output to be
```bash
/Users/project_path/README.md
  1:1  error  Parsing error: Unexpected character '#'

✖ 1 problem (1 error, 0 warnings)
```

**What actually happened? Please include the actual, raw output from ESLint.**
```bash
/Users/project_path/README.md
  1:1  error  Parsing error: Unexpected character '#'

✖ 1 problem (1 error, 0 warnings)

npm ERR! code 1
npm ERR! path /Users/project_path
npm ERR! command failed
npm ERR! command sh -c eslint "README.md"

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/me/.npm/_logs/2021-02-08T02_46_03_273Z-debug.log
```
It looks like some part of ESLint is also doing a `sh -c eslint "README.md"` inside the command. However, `eslint` isn't a command that can run by itself since I installed it locally using npm. It has to be run using `npx eslint`.

This issue doesn't occur when I run ESLint on a file without any errors/warnings.

**Are you willing to submit a pull request to fix this bug?**
I unfortunately don't know what is causing the issue or how to fix this bug.

**Comment by mdjermanovic** [comment]
ESLint doesn't run `sh` command, this seems to be related to `npx`.

> ```shell
> /Users/project_path/README.md
>   1:1  error  Parsing error: Unexpected character '#'
> 
> ✖ 1 problem (1 error, 0 warnings)
> 
> npm ERR! code 1
> npm ERR! path /Users/project_path
> npm ERR! command failed
> npm ERR! command sh -c eslint "README.md"
> ```
> This issue doesn't occur when I run ESLint on a file without any errors/warnings.

ESLint returns exit code 1 if there are errors, so the command fails. This output with `npm ...` looks unusual for `npx` though, but maybe the version you have works like that on macOS?

> npm ERR!     /Users/me/.npm/_logs/2021-02-08T02_46_03_273Z-debug.log

Can you check this log to see what happened?

**Reply by jeffhou** [reply]
This is what the log file output.

My guess is that when there's an error code, `npx` tries to find the globally installed command instead. Is this reproducible on a fresh local install if ESLint was never globally installed? I just followed the instructions to install locally this section of the docs -- https://eslint.org/docs/user-guide/getting-started#installation-and-usage.

FWIW, the same issue happens with any javascript file, e.g. `index.js`. I just used a toy markdown file to illustrate the error.

```
0 verbose cli [
0 verbose cli   '/usr/local/bin/node',
0 verbose cli   '/usr/local/lib/node_modules/npm/bin/npm-cli.js',
0 verbose cli   'exec',
0 verbose cli   '--',
0 verbose cli   'eslint',
0 verbose cli   'README.md'
0 verbose cli ]
1 info using npm@7.5.2
2 info using node@v14.15.3
3 timing config:load:defaults Completed in 2ms
4 timing config:load:file:/usr/local/lib/node_modules/npm/npmrc Completed in 2ms
5 timing config:load:builtin Completed in 2ms
6 timing config:load:cli Completed in 1ms
7 timing config:load:env Completed in 0ms
8 timing config:load:file:/Users/me/projectPath/.npmrc Completed in 0ms
9 timing config:load:project Completed in 1ms
10 timing config:load:file:/Users/me/.npmrc Completed in 0ms
11 timing config:load:user Completed in 0ms
12 timing config:load:file:/usr/local/etc/npmrc Completed in 0ms
13 timing config:load:global Completed in 0ms
14 timing config:load:cafile Completed in 0ms
15 timing config:load:validate Completed in 0ms
16 timing config:load:setUserAgent Completed in 1ms
17 timing config:load:setEnvs Completed in 0ms
18 timing config:load Completed in 7ms
19 verbose npm-session db580581b00d29e7
20 timing npm:load Completed in 15ms
21 timing command:exec Completed in 562ms
22 verbose stack Error: command failed
22 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/index.js:64:27)
22 verbose stack     at ChildProcess.emit (events.js:315:20)
22 verbose stack     at maybeClose (internal/child_process.js:1048:16)
22 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)
23 verbose pkgid lynxwood-vue@0.2.0
24 verbose cwd /Users/me/projectPath
25 verbose Darwin 20.2.0
26 verbose argv "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/bin/npm-cli.js" "exec" "--" "eslint" "README.md"
27 verbose node v14.15.3
28 verbose npm  v7.5.2
29 error code 1
30 error path /Users/me/projectPath
31 error command failed
32 error command sh -c eslint "README.md"
33 verbose exit 1
```

**Reply by mdjermanovic** [reply]
The log doesn't have something like "eslint not found". What would happen if you run `sh -c eslint "README.md"` directly? Maybe `npx eslint` runs `sh -c eslint` right away, not as a fallback.

Can you open an issue on https://github.com/npm/cli repository? They'll certainly know better than us what's actually happening when `npx` is run on your OS, and this doesn't look like something that could be fixed in ESLint (aside from not returning a non-zero exit code, but we shouldn't change that behavior).

**Comment by jeffhou** [comment]
Interesting... I'll look into what's going on some more on my local machine and then file with npm/cli. Probably will do the following:

- Try to reproduce this for other non-eslint modules using `npx`.
- Try to reproduce this on a fresh install of eslint + npm on the same computer (running on Macbook Air M1, zsh).
- Try to reproduce this on a separate OS.

It definitely makes sense to not change the behavior around the error code. Since it's most likely not an issue with eslint, feel free to close this issue.

I encountered this while working on a personal project for the sake of learning eslint, so I may not get to fully troubleshooting/investigating. Thanks for looking into this!
